client/allocrunner/taskrunner/task_runner.go

```diff
@@ -193,6 +193,16 @@ type TaskRunner struct {
 	// maxEvents is the capacity of the TaskEvents on the TaskState.
 	// Defaults to defaultMaxEvents but overrideable for testing.
 	maxEvents int
+
+	// serversContactedCh is passed to TaskRunners so they can detect when
+	// servers have been contacted for the first time in case of a failed
+	// restore.
+	serversContactedCh <-chan struct{}
```

Comment:
This is more precisely: fetched expected running allocations from server - not heartbeat/registration/etc. Not sure if it's worth making the distinction or where to add it.

Reply:
Yeah, we've never had good nomenclature for this. The code base often talks about "syncing" allocs, but since `client->server` and `server->client` updates are completely independent code paths it's always ambiguous to say "syncing".

* `syncedFromServersCh`
* `allocsPulledCh`
* `allocsRetrievedCh`
* `unblockedRestoredCh` - we could go with a name related to its _use_ instead of the event/state it represents, but that doesn't seem as good.

Any thoughts/preferences? `allocsPulledCh` might be my preference as we use the term `pulled` in related code/logging.

Reply:
I'd be ok with clarifying in comment with keeping the field name the same.